components/BottomNavigation/src/private/MDCBottomNavigationLargeItemDialogView.m

```diff
@@ -0,0 +1,133 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCBottomNavigationLargeItemDialogView.h"
+
+#import <CoreGraphics/CoreGraphics.h>
+
+static const CGFloat kTitleFontScaling = (CGFloat)0.5;
+static const CGFloat kTitleFontSize = 35;
+static const CGFloat kImageViewHeightToDialogRatio = (CGFloat)0.35;
+
+/**
+ * Returns an image representing the given UITabBarItem. If the item has a largeContentSizeImage,
+ * this function will return that image, otherwise it returns the value of the item's image
+ * property. This value may be nil.
+ */
+static UIImage *_Nullable MDCImageForItem(UITabBarItem *_Nullable item) {
+  if (!item) {
+    return nil;
+  }
+
+  UIImage *image;
+  if (@available(iOS 11, *)) {
+    image = item.largeContentSizeImage;
+  }
+
+  return image ?: item.image;
+}
+
+@interface MDCBottomNavigationLargeItemDialogView ()
+
+/** The view that hosts the image of the tab bar item. */
+@property(nonatomic, nonnull) UIImageView *imageView;
+
+/** The view that is set to the title of the tab bar item. */
+@property(nonatomic, nonnull) UILabel *titleLabel;
+
+@end
+
+@implementation MDCBottomNavigationLargeItemDialogView
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    [self commonInitMDCBottomNavigationLargeItemView];
+  }
+
+  return self;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    [self commonInitMDCBottomNavigationLargeItemView];
+  }
+
+  return self;
+}
+
+- (void)commonInitMDCBottomNavigationLargeItemView {
+  UIColor *contentColor = [UIColor colorWithWhite:(CGFloat)0.15 alpha:1];
+
+  _imageView = [[UIImageView alloc] init];
+  _imageView.tintColor = contentColor;
+  _imageView.contentMode = UIViewContentModeScaleAspectFit;
+  [self.contentView addSubview:_imageView];
+
+  _titleLabel = [[UILabel alloc] init];
+  _titleLabel.textAlignment = NSTextAlignmentCenter;
+  _titleLabel.lineBreakMode = NSLineBreakByTruncatingTail;
+  _titleLabel.adjustsFontSizeToFitWidth = YES;
+  _titleLabel.minimumScaleFactor = kTitleFontScaling;
+  _titleLabel.textColor = contentColor;
+  _titleLabel.font = [UIFont systemFontOfSize:kTitleFontSize];
+  _titleLabel.numberOfLines = (_imageView.image) ? 1 : 0;
```

Comment:
Is this the behavior of UITabBarController? Since MDCBottomNavigationBar supports multiple lines I want to make sure we're intentionally removing that capability here.

Reply:
If there is an image, there will be only one line and the font will be sized to fit up.

Because `MDCBottomNavigationBar` supports multi-line I changed this to:
```
_titleLabel.numberOfLines = 0;
```